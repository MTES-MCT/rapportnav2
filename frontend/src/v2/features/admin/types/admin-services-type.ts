export interface AdminService {
  id: number
  name: String
  createdAt?: Date
  updatedAt?: Date
  controlUnitId?: number
}

export type AdminAction = 'DELETE' | 'CREATE' | 'UPDATE'
export type AdminCell = { key: string; label: string; width: number }
