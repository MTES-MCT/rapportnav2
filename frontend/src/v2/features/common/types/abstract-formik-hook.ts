import { FormikErrors } from 'formik'
import { ObjectSchema } from 'yup'

export interface AbstractFormikHook<T, M> {
  initValue?: M
  errors?: FormikErrors<M>
  beforeInitValue: (value?: T) => M | undefined
  beforeSubmit: (value?: M, errors?: FormikErrors<M>) => T | undefined
  handleSubmit: (
    value?: M,
    errors?: FormikErrors<M>,
    onSubmit?: (valueToSubmit?: T) => Promise<unknown>
  ) => Promise<void>
}

export interface AbstractFormikSubFormHook<M> {
  initValue?: M
  errors?: FormikErrors<M>
  validationSchema?: ObjectSchema<any>
  handleSubmit: (value?: M, errors?: FormikErrors<M>) => Promise<void>
}
