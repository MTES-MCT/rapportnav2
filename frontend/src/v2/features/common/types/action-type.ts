export enum ActionType {
  CONTROL = 'CONTROL',
  NOTE = 'NOTE',
  SURVEILLANCE = 'SURVEILLANCE',
  STATUS = 'STATUS',
  CONTACT = 'CONTACT',
  RESCUE = 'RESCUE',
  OTHER = 'OTHER',
  NAUTICAL_EVENT = 'NAUTICAL_EVENT',
  VIGIMER = 'VIGIMER',
  ANTI_POLLUTION = 'ANTI_POLLUTION',
  BAAEM_PERMANENCE = 'BAAEM_PERMANENCE',
  PUBLIC_ORDER = 'PUBLIC_ORDER',
  REPRESENTATION = 'REPRESENTATION',
  ILLEGAL_IMMIGRATION = 'ILLEGAL_IMMIGRATION',
  INQUIRY = 'INQUIRY',
  COMMUNICATION = 'COMMUNICATION',
  TRAINING = 'TRAINING',
  UNIT_MANAGEMENT_PLANNING = 'UNIT_MANAGEMENT_PLANNING',
  UNIT_MANAGEMENT_TRAINING = 'UNIT_MANAGEMENT_TRAINING',
  UNIT_MANAGEMENT_OTHER = 'UNIT_MANAGEMENT_OTHER',
  CONTROL_SECTOR = 'CONTROL_SECTOR',
  CONTROL_NAUTICAL_LEISURE = 'CONTROL_NAUTICAL_LEISURE',
  CONTROL_SLEEPING_FISHING_GEAR = 'CONTROL_SLEEPING_FISHING_GEAR',
  OTHER_CONTROL = 'OTHER_CONTROL',
  RESOURCES_MAINTENANCE = 'RESOURCES_MAINTENANCE',
  MEETING = 'MEETING',
  PV_DRAFTING = 'PV_DRAFTING',
  HEARING_CONDUCT = 'HEARING_CONDUCT',
  LAND_SURVEILLANCE = 'LAND_SURVEILLANCE',
  FISHING_SURVEILLANCE = 'FISHING_SURVEILLANCE',
  MARITIME_SURVEILLANCE = 'MARITIME_SURVEILLANCE'
}

export enum ActionGroupType {
  OTHER_GROUP = 'OTHER_GROUP',
  CONTROL_GROUP = 'CONTROL_GROUP',
  SURVEILLANCE_GROUP = 'SURVEILLANCE_GROUP',
  ADMINISTRATIVE_GROUP = 'ADMINISTRATIVE_GROUP',
  UNIT_MANAGEMENT_GROUP = 'UNIT_MANAGEMENT_GROUP'
}
