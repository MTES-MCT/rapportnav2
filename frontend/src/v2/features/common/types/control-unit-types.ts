export type ControlUnitResource = {
  id: string
  controlUnit: ControlUnit
  controlUnitId: number
  isArchived: boolean
  name: string
  note?: string
  photo?: string
  station: Station
  stationId: number
  type: ControlUnitResourceType
}

export type Station = {
  id: number
  latitude: number
  longitude: number
  name: string
}

export type ControlUnit = {
  id: number
  areaNote?: string
  administrationId: number
  departmentAreaInseeCode?: string
  isArchived: boolean
  name: string
  termsNote?: string
}

export type Administration = {
  id: number
  name: string
  controlUnitIds: number[]
  isArchived: boolean
  controlUnits: ControlUnit[]
}

enum ControlUnitResourceType {
  AIRPLANE = "AIRPLANE",
  BARGE = "BARGE",
  CAR = "CAR",
  DRONE = "DRONE",
  EQUESTRIAN = "EQUESTRIAN",
  FAST_BOAT = "FAST_BOAT",
  FRIGATE = "FRIGATE",
  HELICOPTER = "HELICOPTER",
  HYDROGRAPHIC_SHIP = "HYDROGRAPHIC_SHIP",
  KAYAK = "KAYAK",
  LIGHT_FAST_BOAT = "LIGHT_FAST_BOAT",
  MINE_DIVER = "MINE_DIVER",
  MOTORCYCLE = "MOTORCYCLE",
  NET_LIFTER = "NET_LIFTER",
  NO_RESOURCE = "NO_RESOURCE",
  OTHER = "OTHER",
  PATROL_BOAT = "PATROL_BOAT",
  PEDESTRIAN = "PEDESTRIAN",
  PIROGUE = "PIROGUE",
  RIGID_HULL = "RIGID_HULL",
  SEA_SCOOTER = "SEA_SCOOTER",
  SEMI_RIGID = "SEMI_RIGID",
  SUPPORT_SHIP = "SUPPORT_SHIP",
  TRAINING_SHIP = "TRAINING_SHIP",
  TUGBOAT = "TUGBOAT"
}
