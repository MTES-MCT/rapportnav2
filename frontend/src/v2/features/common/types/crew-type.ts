import { Service } from '@common/types/crew-types.ts'

export type Agent = {
  id: number
  firstName: string
  lastName: string
  services: Service[]
}

export type AgentRole = {
  id: string
  title: string
}

export type MissionCrewMember = {
  id?: string
  agent: Agent
  comment?: string
  role?: AgentRole
  missionId?: number
}
