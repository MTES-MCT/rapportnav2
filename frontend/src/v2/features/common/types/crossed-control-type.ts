export enum CrossControlOriginType {
  OPPORTUNITY_CONTROL = 'OPPORTUNITY_CONTROL',
  FOLLOW_UP_CONTROL = 'FOLLOW_UP_CONTROL',
  CNSP_REPORTING = 'CNSP_REPORTING',
  OTHER_REPORTING = 'OTHER_REPORTING',
  URCEM_DEDICATED_STATION = 'URCEM_DEDICATED_STATION'
}

export enum CrossControlStatusType {
  NEW = 'NEW',
  FOLLOW_UP = 'FOLLOW_UP',
  CLOSED = 'CLOSED'
}

export enum CrossControlTargetType {
  VEHICLE = 'VEHICLE',
  COMPANY = 'COMPANY'
}

export enum CrossControlConclusionType {
  NO_FOLLOW_UP = 'NO_FOLLOW_UP',
  WITH_REPORT = 'WITH_REPORT'
}

export interface CrossControl {
  id?: string
  agentId?: number
  vesselId?: number
  serviceId?: number
  nbrOfHours?: number
  endDateTimeUtc?: string
  startDateTimeUtc?: string
  isSignedByInspector?: boolean
  type?: CrossControlTargetType
  origin?: CrossControlOriginType
  status?: CrossControlStatusType
  conclusion?: CrossControlConclusionType
  vesselName?: string
  isReferentialClosed?: boolean
  vesselExternalReferenceNumber?: string
}
