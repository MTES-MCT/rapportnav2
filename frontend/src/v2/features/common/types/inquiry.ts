import { MissionAction } from './mission-action'

export enum InquiryOriginType {
  OPPORTUNITY_CONTROL = 'OPPORTUNITY_CONTROL',
  FOLLOW_UP_CONTROL = 'FOLLOW_UP_CONTROL',
  CNSP_REPORTING = 'CNSP_REPORTING',
  OTHER_REPORTING = 'OTHER_REPORTING',
  URCEM_DEDICATED_STATION = 'URCEM_DEDICATED_STATION'
}

export enum InquiryStatusType {
  NEW = 'NEW',
  CLOSED = 'CLOSED',
  FOLLOW_UP = 'FOLLOW_UP'
}

export enum InquiryConclusionType {
  NO_FOLLOW_UP = 'NO_FOLLOW_UP',
  WITH_REPORT = 'WITH_REPORT'
}

export enum InquiryTargetType {
  VEHICLE = 'VEHICLE',
  COMPANY = 'COMPANY'
}

export interface Inquiry {
  id?: string
  type?: string
  agentId?: number
  vesselId?: number
  serviceId?: number
  endDateTimeUtc?: string
  startDateTimeUtc?: string
  origin?: InquiryOriginType
  status?: InquiryStatusType
  conclusion?: InquiryConclusionType
  vesselName?: string
  isSignedByInspector?: boolean
  vesselExternalReferenceNumber?: string
  missionId?: number
  missionIdUUID?: string
  actions?: MissionAction[]
}
