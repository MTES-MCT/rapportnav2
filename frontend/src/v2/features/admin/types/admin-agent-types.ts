import { AdminService } from './admin-services-type'

export type Agent = {
  id?: string
  firstName?: string
  lastName?: string
  services?: string[]
  deletedAt?: Date
  createdAt?: Date
  updatedAt?: Date
}

export type AgentRole = {
  id: string
  title: string
}

export type AdminAgentService = {
  id: number
  agent: Agent
  role: AgentRole
  createdAt?: Date
  updatedAt?: Date
  disabledAt?: Date
}

export type AdminServiceWithAgent = {
  service: AdminService
  agentServices: AdminAgentService[]
}

export type AdminAgentServiceInput = {
  id?: number
  agentId: number
  serviceId: number
  roleId?: number
}
