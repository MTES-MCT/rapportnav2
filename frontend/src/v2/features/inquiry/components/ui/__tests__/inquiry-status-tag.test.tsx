import { describe, expect, it } from 'vitest'
import { InquiryStatusType } from '../../../../common/types/inquiry'
import InquiryStatusTag from '../inquiry-status-tag'
import { render } from './../../../../../../test-utils'

describe('InquiryStatusTag', () => {
  it('renders should mathc snapshot', () => {
    const wrapper = render(<InquiryStatusTag status={InquiryStatusType.IN_PROGRESS} />)
    expect(wrapper).toMatchSnapshot()
  })
})
