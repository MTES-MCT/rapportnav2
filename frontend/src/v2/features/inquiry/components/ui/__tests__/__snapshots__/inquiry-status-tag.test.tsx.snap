// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InquiryStatusTag > renders should mathc snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-fOOuSg hzVeCO Element-Tag"
      >
        <span
          class="sc-hjsuWn jbkALq Element-IconBox"
        >
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
          >
            <g
              transform="translate(240 -41)"
            >
              <g>
                <g
                  fill="none"
                  stroke-miterlimit="10"
                >
                  <path
                    d="M-230,42a9,9,0,0,0-9,9,9,9,0,0,0,9,9,9,9,0,0,0,9-9A9,9,0,0,0-230,42Zm0,16a7,7,0,0,1-7-7,7,7,0,0,1,7-7,7,7,0,0,1,7,7A7,7,0,0,1-230,58Z"
                    stroke="none"
                  />
                  <path
                    d="M -230 42 C -225.0290069580078 42 -221 46.02899932861328 -221 51 C -221 55.97100067138672 -225.0290069580078 60 -230 60 C -234.9709930419922 60 -239 55.97100067138672 -239 51 C -239 46.02899932861328 -234.9709930419922 42 -230 42 Z M -230 58 C -226.1340026855469 58 -223 54.86600112915039 -223 51 C -223 47.13399887084961 -226.1340026855469 44 -230 44 C -233.8659973144531 44 -237 47.13399887084961 -237 51 C -237 54.86600112915039 -233.8659973144531 58 -230 58 Z"
                    fill="currentColor"
                    stroke="none"
                  />
                </g>
                <g
                  fill="none"
                  stroke-miterlimit="10"
                >
                  <path
                    d="M-229,50V46h-2v6h6V50Z"
                    stroke="none"
                  />
                  <path
                    d="M -225 52 L -231 52 L -231 46 L -229 46 L -229 50 L -225 50 L -225 52 Z"
                    fill="currentColor"
                    stroke="none"
                  />
                </g>
              </g>
              <rect
                fill="none"
                height="20"
                transform="translate(-240 41)"
                width="20"
              />
            </g>
          </svg>
        </span>
        <p
          class="sc-iRcyzz sc-fFmWCW hlWJTw hAFmCO"
          color="#3B4559"
          font-size="13"
          weight="medium"
        >
          En cours
        </p>
      </span>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-fOOuSg hzVeCO Element-Tag"
    >
      <span
        class="sc-hjsuWn jbkALq Element-IconBox"
      >
        <svg
          height="20"
          viewBox="0 0 20 20"
          width="20"
        >
          <g
            transform="translate(240 -41)"
          >
            <g>
              <g
                fill="none"
                stroke-miterlimit="10"
              >
                <path
                  d="M-230,42a9,9,0,0,0-9,9,9,9,0,0,0,9,9,9,9,0,0,0,9-9A9,9,0,0,0-230,42Zm0,16a7,7,0,0,1-7-7,7,7,0,0,1,7-7,7,7,0,0,1,7,7A7,7,0,0,1-230,58Z"
                  stroke="none"
                />
                <path
                  d="M -230 42 C -225.0290069580078 42 -221 46.02899932861328 -221 51 C -221 55.97100067138672 -225.0290069580078 60 -230 60 C -234.9709930419922 60 -239 55.97100067138672 -239 51 C -239 46.02899932861328 -234.9709930419922 42 -230 42 Z M -230 58 C -226.1340026855469 58 -223 54.86600112915039 -223 51 C -223 47.13399887084961 -226.1340026855469 44 -230 44 C -233.8659973144531 44 -237 47.13399887084961 -237 51 C -237 54.86600112915039 -233.8659973144531 58 -230 58 Z"
                  fill="currentColor"
                  stroke="none"
                />
              </g>
              <g
                fill="none"
                stroke-miterlimit="10"
              >
                <path
                  d="M-229,50V46h-2v6h6V50Z"
                  stroke="none"
                />
                <path
                  d="M -225 52 L -231 52 L -231 46 L -229 46 L -229 50 L -225 50 L -225 52 Z"
                  fill="currentColor"
                  stroke="none"
                />
              </g>
            </g>
            <rect
              fill="none"
              height="20"
              transform="translate(-240 41)"
              width="20"
            />
          </g>
        </svg>
      </span>
      <p
        class="sc-iRcyzz sc-fFmWCW hlWJTw hAFmCO"
        color="#3B4559"
        font-size="13"
        weight="medium"
      >
        En cours
      </p>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
